config/jobs/kubernetes-csi/volume-data-source-validator/volume-data-source-validator-config.yaml

```diff
@@ -24,11 +25,355 @@ presubmits:
         args:
         - ./.prow.sh
         env:
+        # We pick some version for which there are pre-built images for kind.
+        # Update only when the newer version is known to not cause issues,
+        # otherwise presubmit jobs may start to fail for reasons that are
+        # unrelated to the PR. Testing against the latest Kubernetes is covered
+        # by periodic jobs (see https://k8s-testgrid.appspot.com/sig-storage-csi-ci#Summary).
+        - name: CSI_PROW_KUBERNETES_VERSION
+          value: "1.21.0"
+        - name: CSI_PROW_USE_BAZEL
+          value: "false"
+        - name: CSI_PROW_KUBERNETES_DEPLOYMENT
+          value: "1.21"
+        - name: CSI_PROW_DEPLOYMENT_SUFFIX
+          value: ""
+        - name: CSI_PROW_DRIVER_VERSION
+          value: "v1.7.2"
         - name: CSI_SNAPSHOTTER_VERSION
-          value: "v3.0.3"
+          value: "v4.0.0"
+        - name: CSI_PROW_TESTS
+          value: "sanity serial parallel"
         # docker-in-docker needs privileged mode
         securityContext:
           privileged: true
         resources:
           requests:
             cpu: 2000m
+  - name: pull-kubernetes-csi-volume-data-source-validator-1-21-on-kubernetes-master
+    # Explicitly needs to be started with /test.
+    # This cannot be enabled by default because there's always the risk
+    # that something changes in master which breaks the pre-merge check.
+    always_run: false
+    optional: true
+    decorate: true
+    skip_report: false
+    labels:
+      preset-service-account: "true"
+      preset-dind-enabled: "true"
+      preset-bazel-remote-cache-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    annotations:
+      testgrid-dashboards: sig-storage-csi-volume-data-source-validator
+      testgrid-tab-name: 1-21-on-kubernetes-master
+      description: Kubernetes-CSI pull job in repo volume-data-source-validator for non-alpha tests, using deployment 1.21 on Kubernetes master
+    spec:
+      containers:
+      # We need this image because it has Docker in Docker and go.
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220315-d8db32377d-master
+        command:
+        - runner.sh
+        args:
+        - ./.prow.sh
```

Comment:
Do we also need to update [prow.sh in release-tools repo](https://github.com/kubernetes-csi/external-snapshotter/blob/master/release-tools/prow.sh#L788-L793) or [prow.sh in volume-data-source-validator repo](https://github.com/kubernetes-csi/volume-data-source-validator/blob/master/.prow.sh) so that the container image built for the change in the volume-data-source-validator repo is used in the test?

Note that e2e test for volume-data-source-validator is deployed inside each test [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/provisioning.go#L268-L276). So, we will need to consider a way to replace the image for this implementation. 

(I hope that just `kind load docker-image` the newly built image with the same name specified in "test/e2e/testing-manifests/storage-csi/any-volume-datasource/crd/populator.storage.k8s.io_volumepopulators.yaml" works. And such a script code would be better added to prow.sh in volume-data-source-validator repo, not release-tools repo).

Reply:
I have created a PR to deploy volume data source validators with the prow deployment: https://github.com/kubernetes-csi/volume-data-source-validator/pull/37

Not sure how to address changing the image tag in the e2e tests [here](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/provisioning.go#L268-L276) as it is hardcoded. Would appreciate any feedback on the PR.